lsp/src/main/java/org/lflang/diagram/lsp/LFLanguageServerExtension.java

```diff
@@ -84,6 +100,82 @@ public CompletableFuture<String> build(BuildArgs args) {
+   * {@code filePath}.
+   *
+   * @param filePath the URI of the LF file of interest
+   * @return A message describing the outcome of the build process.
+   */
+  @JsonRequest("generator/getLibraryReactors")
+  public CompletableFuture<LibraryFileTree> getLibraryReactors(String filePath) {
+    return CompletableFuture.supplyAsync(
+        () -> {
+          try {
+            // LF program file parsing
+            URI uri = URI.createURI(filePath);
+            // Return a list of reactors within the file at the specific uri
+            return parseLibraryReactors(uri);
+          } catch (Exception e) {
+            return null;
+          }
+        });
+  }
+
+  /**
+   * Retrieves the target position specified in the LF program file at the given path.
+   *
+   * @param path The path to the LF program file.
+   * @return A CompletableFuture containing the NodePosition object representing the position of the
+   *     target, or null if an error occurs during parsing or if the target position is not found.
+   */
+  @JsonRequest("generator/getTargetPosition")
+  public CompletableFuture<NodePosition> getTargetPosition(String path) {
+    return CompletableFuture.supplyAsync(
+        () -> {
+          NodePosition targetPosition = null;
+          try {
+            URI uri = URI.createURI(path);
+            // LF program file parsing
+            Resource resource = getXtextResourceSet(uri).getResource(uri, true);
+            Model m = (Model) resource.getContents().get(0);
+            TargetDecl target = m.getTarget();
+            INode node = NodeModelUtils.getNode(target);
+            targetPosition = new NodePosition(node.getStartLine(), node.getEndLine());
+            return targetPosition;
+          } catch (Exception e) {
+            return null;
+          }
+        });
+  }
+
+  /**
+   * Parses a library of reactors specified by the provided URI and constructs a hierarchical
+   * libraryFileTree representation.
+   *
+   * @param uri The URI specifying the location of the library.
+   * @return A Tree object representing the hierarchical structure of the reactor library, or null
+   *     if an error occurs during parsing.
+   */
+  public LibraryFileTree parseLibraryReactors(URI uri) {
+    LibraryFileTree res = new LibraryFileTree(uri.toString());
+    try {
+      Resource resource = getXtextResourceSet(uri).getResource(uri, true);
+      Model m = (Model) resource.getContents().get(0);
+      Stream<Reactor> reactors =
+          m.getReactors().stream().filter(r -> r.getName() != null && !r.getName().isEmpty());
+      reactors.forEach(
+          r -> {
+            INode node = NodeModelUtils.getNode(r);
+            NodePosition nodePosition = new NodePosition(node.getStartLine(), node.getEndLine());
+            res.addChild(new LibraryFileTreeNode(r.getName(), res.getUri(), nodePosition));
+          });
+    } catch (Exception e) {
+      return null;
```

Comment:
Should this print a warning?  I'm not sure how VS code would handle this output, but it seems possibly better than silently failing.

Reply:
You're right, I could add a warning message and display it as a popup in VSCode. This should work.

Reply:
Have you considered not having a pop-up and logging the warning instead?
https://code.visualstudio.com/api/ux-guidelines/notifications

Reply:
Thank you, @soerendomroes, I'll check it out

Reply:
@soerendomroes I've handled this situation in the following way:

1. I send the request from the client (VSCode extension).
2. I receive and process the request in the Language Server.
3. If an error occurs, I'll return `null` from the Language Server.
4. If the object request received by the client is `null`, I show an error message by executing the following code:

    ```typescript
    vscode.window.showErrorMessage('Error retrieving data from the Language Server', 
    ...['Try again', 'Cancel']).then(selection => {
        if (selection === 'Try again') {
            this.refreshTree(); // try again
        }
    });
    ```

Does this sound like a good solution?